### PR TITLE
Prevent deletion of non-orphaned Argo CD deployed resources 

### PR DESCRIPTION
**Pull request by jgwest**
#### Description:
- Resources that were not orphaned are being deleted, due to a bug in parsing the application value
- Updates to unit tests to fix test issues that should have identified the above problem.

### CODE REVIEW

**Comment by jgwest** [comment]
CI fail on known flake, for which there is a GITOPSRVCE- issue open.

/retest

File: Makefile [hunk 1]
```diff
@@ -251,6 +251,16 @@ lint: ## Run lint checks for all components
...
 	cd $(MAKEFILE_ROOT)/utilities/init-container && make lint
 
+gosec: ## Run gosec checks for all components
+	cd $(MAKEFILE_ROOT)/backend-shared && make gosec
+	cd $(MAKEFILE_ROOT)/backend && make gosec
+	cd $(MAKEFILE_ROOT)/cluster-agent && make gosec
+	cd $(MAKEFILE_ROOT)/appstudio-controller && make gosec
+	cd $(MAKEFILE_ROOT)/tests-e2e && make gosec
+#	cd $(MAKEFILE_ROOT)/utilities/db-migration && make gosec
+#	cd $(MAKEFILE_ROOT)/utilities/init-container && make gossec
```
Comment (chetan-rns) [inline Makefile:261]: Are these comments intentional? Should we check gosec for utilities as well?

**Review by chetan-rns — APPROVED** [review]
Thanks for fixing the bug @jgwest. The PR looks good. I have added a comment about checking for gosec in utilities.

/lgtm

File: Makefile (hunk 1 above)
Comment (jgwest) [reply Makefile:261]: Thanks Chetan, this was mostly a placeholder so we remember to add gosec to these in the future.

**Pull request merged**